templates/explore/repo_search.tmpl

```diff
@@ -20,8 +20,21 @@
 			<a class="{{if eq .SortType "fewestforks"}}active {{end}}item" href="{{$.Link}}?sort=fewestforks&q={{$.Keyword}}&language={{$.Language}}">{{.locale.Tr "repo.issues.filter_sort.fewestforks"}}</a>
 		</div>
 	</div>
+
+	<!-- Language -->
+	<div class="ui right dropdown type jump item">
+		<span class="text">Language {{svg "octicon-triangle-down" 14 "dropdown icon"}}</span>
+		<div class="menu">
+			<a class="item" href="{{$.Link}}?q={{$.Keyword}}&sort={{$.SortType}}">Any</a>
+			{{range $lang, $color := $.Languages}}
+			<a class="{{if eq $.Language $lang}}active {{end}}item" href="{{$.Link}}?q={{$.Keyword}}&language={{$lang}}&sort={{$.SortType}}">
+				<span class="gt-df gt-ac gt-mr-3"><i class="color-icon gt-mr-3" style="background-color: {{$color}}"></i>{{$lang}}</span>
+			</a>
+			{{end}}
+		</div>
+	</div>
```

Comment:
Is this really suitable? The big Problem I was facing, was that this list can get extremely large, so I needed to find another solution. I tried to rewrite the Author search Box from the Issue page for languages, but I don't got it to work.

Reply:
Now that I think about it, this probably isn't the right way to implement this :/

Reply:
Here's my Go function of collectiong all Languages used in the Instance:
```go
func GetPrimaryRepoLanguageList(ctx context.Context) (LanguageStatList, error) {
	languageList := make(LanguageStatList, 0)

	err := db.GetEngine(ctx).Table("language_stat").Cols("language").Where(builder.Eq{"is_primary": true}).Distinct("language").OrderBy("language").Find(&languageList)

	if err != nil {
		return nil, err
	}

	languageList.LoadAttributes()
	return languageList, nil
}
```
Put it in `models/repo/repo_list.go`.

> I felt that computing the list of languages for all the repos (while insignificant for small instances) might potentially be a bottleneck for larger instances

I have no Way to test this on a very large instance, but considering the language Column is indexed, it should not be that big of a deal. It is still better than showing a List of all Languages where in most cases not more than 10 or so are used.

Reply:
Thank you!